components/EmotionForm.js

```diff
@@ -0,0 +1,189 @@
+  const currentDateTime = new Date(
+    new Date().getTime() - new Date().getTimezoneOffset() * 60000
+  )
+    .toISOString()
+    .slice(0, 16);
+
+  const [selectedEmotion, setSelectedEmotion] = useState("");
+  const [selectedIntensity, setSelectedIntensity] = useState(5);
+  const [selectedDateTime, setSelectedDateTime] = useState(currentDateTime);
+  const [notes, setNotes] = useState("");
+  const [formError, setFormError] = useState(""); // Error message if not all fields are filled out on submit
+  const [successMessage, setSuccessMessage] = useState(""); // Success message
+  const [hasError, setHasError] = useState(false); // New state variable for styling
+
+  // Change emotion
+  function handleEmotionChange(event) {
+    setSelectedEmotion(event.target.value);
+    if (event.target.value !== "") {
+      setFormError(""); // Reset error message when the field is filled
+      setHasError(false); // Reset styling once an emotion is selected
+    }
+  }
+
+  // Change intensity
+  function handleIntensityChange(event) {
+    setSelectedIntensity(event.target.value);
+  }
+
+  // Change dateTime
+  function handleDateTimeChange(event) {
+    setSelectedDateTime(event.target.value);
+    if (event.target.value) {
+      setFormError(""); // Reset error message when date/time is selected
+    }
+  }
+
+  // Save notes
+  function handleNotesChange(event) {
+    setNotes(event.target.value);
+  }
+
+  // Trigger submit
+  function handleSubmit(event) {
+    event.preventDefault();
+    // Validate: Emotion must be selected
+    if (!selectedEmotion) {
+      setFormError("Please fill in the required fields.");
+      setHasError(true); // Set error state for styling
+      setSuccessMessage(""); // Reset success message
+      return;
+    }
+    // Validate: DateTime must be set
+    if (!selectedDateTime) {
+      setFormError("Please select a date and time.");
+      setSuccessMessage(""); // Reset success message
+      return;
+    }
+
+    // New object
+    const newEmotionEntry = {
+      emotion: selectedEmotion,
+      intensity: selectedIntensity,
+      dateTime: selectedDateTime,
+      notes: notes,
+    };
+    // Pass new object
+    onCreateEmotion(newEmotionEntry);
+
+    // Reset form
+    setSelectedEmotion("");
+    setSelectedIntensity(5);
+    setSelectedDateTime(currentDateTime);
+    setNotes("");
+    setFormError(""); // Reset error message
+    setHasError(false); // Reset error state for styling
+    setSuccessMessage("Emotion successfully added!"); // Set success message
+    // Reset success message after 5 seconds
+    setTimeout(() => {
+      setSuccessMessage("");
+    }, 5000);
+  }
+
+  return (
+    <>
+      <StyledSubheadline>Add new Emotion</StyledSubheadline>
+      <StyledEmotionForm onSubmit={handleSubmit}>
+        {/* Dynamic label for emotion, red only after form submission if no emotion is selected */}
+        <StyledLabel htmlFor="emotion" hasError={hasError}>
```

Comment:
You should prefix props you pass to a styled component with "$". Otherwise you will get an error message in your terminal.

```suggestion
        <StyledLabel htmlFor="emotion" $hasError={hasError}>
```